app/src/main/java/org/mozilla/fenix/composables/ListItemTabLarge.kt

```diff
@@ -0,0 +1,154 @@
+ * ```
+ * ---------------------------------------------
+ * | -------------- Title                      |
+ * | |    Image   | wrapped on                 |
+ * | |    from    | three rows if needed       |
+ * | |  imageUrl  |                            |
+ * | -------------- Optional caption           |
+ * ---------------------------------------------
+ * ```
+ *
+ * @param client [Client] instance to be used for downloading the image.
+ * When using [GeckoViewFetchClient] the image will automatically be cached if it has the right headers.
+ * @param imageUrl URL from where the to download a header image of the tab this composable renders.
+ * @param title title off the tab this composable renders.
+ * @param caption optional caption text.
+ * @param onClick optional callback to be invoked when this composable is clicked.
+ */
+@Composable
+fun ListItemTabLarge(
+    client: Client,
+    imageUrl: String,
+    title: String,
+    caption: String? = null,
+    onClick: (() -> Unit)? = null
+) {
+    CommonSurface(client, imageUrl, onClick) {
+        TabTitle(text = title, maxLines = 3)
+
+        if (caption != null) {
+            TabSubtitle(text = caption)
+        }
+    }
+}
+
+/**
+ * Default layout of a large tab shown in a list taking composable arguments for title and caption
+ * allowing as an exception to customize these elements.
+ * Has the following structure:
+ * ```
+ * ---------------------------------------------
+ * | -------------- -------------------------- |
+ * | |            | |         Title          | |
+ * | |    Image   | |       composable       | |
+ * | |    from    | -------------------------- |
+ * | |  imageUrl  | -------------------------- |
+ * | |            | |   Optional composable  | |
+ * | -------------- -------------------------- |
+ * ---------------------------------------------
+ * ```
+ *
+ * @param client [Client] instance to be used for downloading the image.
+ * When using [GeckoViewFetchClient] the image will automatically be cached if it has the right headers.
+ * @param imageUrl URL from where the to download a header image of the tab this composable renders.
+ * @param title composable rendering the title of the tab this composable represents.
+ * @param subtitle optional tab caption composable.
+ * @param onClick optional callback to be invoked when this composable is clicked.
+ */
+@Composable
+fun ListItemTabLarge(
+    client: Client,
+    imageUrl: String,
+    onClick: () -> Unit,
+    title: @Composable () -> Unit,
+    subtitle: @Composable (() -> Unit)? = null
+) {
+    CommonSurface(client, imageUrl, onClick) {
+        title()
+
+        subtitle?.invoke()
+    }
+}
+
+/**
+ * Shared default configuration of a ListItemTabLarge Composable.
+ */
+@Composable
+private fun CommonSurface(
+    client: Client,
+    imageUrl: String,
+    onClick: (() -> Unit)? = null,
+    tabDetails: @Composable () -> Unit
+) {
+    var modifier = Modifier.size(328.dp, 116.dp)
```

Comment:
I think we don't even need to call this `ListItemTabLarge` if we add a modifier argument to the parameters and allow the caller to specify the size.

Reply:
This is `ListItemTabLarge` as a default for this usecase which for a cohesive design should not change.
The image dimensions are hardcoded and so does the size of this to force all users stick to this design. Wherever this is used it will look the same.
For another usecase we'd have `ListiItemTabMedium` / `ListItemTabSmall`, again them having as many as possible hardcodes to enforce the same design all throughout the app.
More of these variants needed -> clear sign that the design becomes bloated and the UX looses it's cohesiveness.